### PR TITLE
 fmt(Refactor): Refactor istgt_lu_add_unit(ISTGT_Ptr, CF_SECTION) 

### PR DESCRIPTION
**Pull request by kn-neelalohitha**
## What is this PR ?
Modularise istgt_lu_add_unit(ISTGT_Ptr, CF_SECTION) function.
## Which issue this PR fixes?
[#2096](https://github.com/openebs/openebs/issues/2096) of [openebs/istgt](https://github.com/openebs/istgt) repository.

signed off by: knn kn.neelalohitha@gmail.com

### CODE REVIEW

File: src/istgt_lu.c [hunk 1]
```diff
@@ -1369,91 +1369,36 @@ istgt_lu_set_local_settings(ISTGT_Ptr istgt, CF_SECTION *sp, ISTGT_LU_Ptr lu)
 
 extern int g_num_luworkers;
 
-static int
-istgt_lu_add_unit(ISTGT_Ptr istgt, CF_SECTION *sp)
+static int 
+sp_validate(ISTGT_Ptr istgt, ISTGT_LU_Ptr lu, CF_SECTION *sp)
```
Comment (vishnuitta) [inline src/istgt_lu.c]: `lu` is not required here

File: src/istgt_lu.c [hunk 2]
```diff
@@ -2204,6 +2166,179 @@ istgt_lu_add_unit(ISTGT_Ptr istgt, CF_SECTION *sp)
...
+	lu->state = ISTGT_STATE_INVALID;
+#if 0
+	/* disabled now */
+	nbs64 = istgt_lu_get_nbserial(istgt->nodebase);
+	nbs = (int) (nbs64 % 900) * 100000;
+#else
+	nbs = 0;
+#endif
+
+	ret = sp_validate(istgt, lu, sp);
```
Comment (vishnuitta) [inline src/istgt_lu.c]: this can be called before allocation of `lu`

File: src/istgt_lu.c [hunk 3]
```diff
@@ -2204,6 +2166,179 @@ istgt_lu_add_unit(ISTGT_Ptr istgt, CF_SECTION *sp)
...
+	if(ret == -1)
+		return (-1);
+	val = istgt_get_val(sp, "TargetName");
+	if (val == NULL) {
+		ISTGT_ERRLOG("LU%d: TargetName not found\n", lu->num);
+		return (-1);
```
Comment (vishnuitta) [inline src/istgt_lu.c]: memory alloted for `lu` will be leaked here if returns from here

File: src/istgt_lu.c (hunk 1 above)
Comment (kn-neelalohitha) [reply src/istgt_lu.c]: Have made the changes.

**Comment by mittachaitu** [comment]
@kn-neelalohitha can we resolve the conflicts to this PR for further reviews?

**Comment by mittachaitu** [comment]
Closing this PR and we can reopen and rework on it when it is required.

**Pull request closed without merging**